### PR TITLE
Trim vendor prefix from openapi names

### PR DESCRIPTION
**Pull request by liggitt**
xref https://github.com/kubernetes/kubernetes/pull/98027

/cc @mikedanese @apelisse

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ func ToRESTFriendlyName(name string) string {
 // Example for vendored Go type:
 //     Original full path:  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.Pod
 //     Canonical name:      k8s.io/api/core/v1.Pod
+//
+//     Original full path:  vendor/k8s.io/api/core/v1.Pod
```
Comment (apelisse) [inline pkg/util/util.go:84]: It took me way longer than it should do find the difference between these two and the two above :-)

**Review by apelisse — COMMENTED** [review]
is there any way to test that? ;-)

/lgtm
/approve

**Comment by liggitt** [comment]
> is there any way to test that? ;-)

I tried, but that required making a type that lived under a vendor path, which for a repo that doesn't vendor ... was non-trivial

**Comment by apelisse** [comment]
Thanks!
/unhold

**Pull request merged**